src/vcf/meta_entry.cpp

```diff
@@ -182,8 +182,42 @@ namespace ebi
             type_field != "String") {
             throw new MetaSectionError{entry.line, "FORMAT metadata Type is not a Integer, Float, Character or String"};
         }
+
+        if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
+            check_format_predefined_type(value, format_type_v41_v42);
+            check_format_predefined_number(value, format_number_v41_v42);
+        } else {
+            check_format_predefined_type(value, format_type_v43);
+            check_format_predefined_number(value, format_number_v43);
+        }
     }
-    
+
+    void MetaEntryVisitor::check_format_predefined_type(std::map<std::string, std::string> & value,
```

Comment:
The implementations of `check_format_predefined_type` and `check_format_predefined_number` are exactly the same as for the INFO field, only the parameters provided change.

There could be just 2 methods `check_predefined_type` and `check_predefined_number`.